### PR TITLE
Feature/JSON and image mutation

### PR DESCRIPTION
**Pull request by AshakaE**
# Graph API Mutation

_API Endpoint to pin JSON and Image to Pinata -Ipfs server_

## How should this be tested?

1. view doc [here](https://documenter.getpostman.com/view/13541794/UVsMvmHZ)

## Linked issues
[Issue](https://github.com/EveripediaNetwork/issues/issues/120)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,44 +26,47 @@
...
     "@nestjs/platform-express": "^8.1.1",
     "@nestjs/typeorm": "^8.0.3",
+    "@pinata/sdk": "^1.1.23",
     "apollo-server-express": "^3.6.3",
-    "class-transformer": "^0.5.1",
     "class-validator": "^0.13.2",
     "graphql": "^16.3.0",
     "graphql-request": "^4.0.0",
+    "graphql-upload": "^13.0.0",
+    "ipfs-core": "^0.14.1",
```
Comment (kesar) [inline package.json:44]: is it ipfs-core used anywhere?

File: package.json [hunk 2]
```diff
@@ -26,44 +26,47 @@
...
+    "jest-mock": "^27.5.1",
     "mysql2": "^2.3.3",
     "nest-commander": "^2.4.0",
-    "pg": "^8.7.3",
```
Comment (kesar) [inline package.json:44]: why?

can you review the package.json and just update what its needed for this PR? I dont want to introduce collateral bugs related to upgrade packages that are not needed

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "start:prod": "node dist/main",
     "console": "node dist/console.js",
     "console:dev": "ts-node src/console.ts",
-    "typeorm": "ts-node -r tsconfig-paths/register ./node_modules/typeorm/cli.js",
+    "typeorm": "ts-node -r tsconfig-paths/register ./node_modules/typeorm/cli.js --config ./src/Database/ormconfig.ts",
```
Comment (kesar) [inline package.json:18]: why? doesnt work without that?

File: package.json [hunk 4]
```diff
@@ -79,7 +82,7 @@
     "ts-loader": "^9.2.6",
     "ts-node": "^10.4.0",
     "tsconfig-paths": "^3.11.0",
-    "typescript": "^4.6.2"
+    "typescript": "^4.5.5"
```
Comment (kesar) [inline package.json:85]: 4.6.2 should work fine afaik

File: src/App/pinJSONAndImage/model/ipfsHash.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { ObjectType, Field } from '@nestjs/graphql'
+import { IsOptional } from 'class-validator'
+
+@ObjectType()
+export default class IpfsHash {
+  @Field({ nullable: true })
```
Comment (kesar) [inline src/App/pinJSONAndImage/model/ipfsHash.ts:6]: can the hash be null?

File: src/App/pinJSONAndImage/pin.service.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  async pinImage(file: fs.PathLike): Promise<any> {
+    const readableStreamForFile = fs.createReadStream(file)
+
+    const pinImageToPinata = async (option: any) => {
+      this.pinata().pinFileToIPFS(option)
+    }
+
+    // const pinImageToIPFS = async () => {
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.service.ts:26]: please remove comments before getting into PR

File: src/App/pinJSONAndImage/pin.service.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+class PinService {
+  constructor(private configService: ConfigService) {}
+
+  private pinata() {
+    const key = this.configService.get<string>('ApiKey')
+    const secret = this.configService.get<string>('ApiSecret')
+    return pinataSDK(key, secret)
+  }
+
+  async pinImage(file: fs.PathLike): Promise<any> {
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.service.ts:19]: can return always a ipfs string instead of any?

File: src/App/pinJSONAndImage/pin.service.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    try {
+      const res = await pinImageToPinata(readableStreamForFile)
+      return res
+      //   return await pinImageToIPFS()
+    } catch (e) {
+      return e
+    }
+  }
+
+  async pinJSON(body: string): Promise<any> {
```
Comment (kesar) [inline src/App/pinJSONAndImage/pin.service.ts:45]: can return always a ipfs string instead of any?

File: package.json (hunk 1 above)
Comment (AshakaE) [reply package.json:44]: I'm not sure of the condition to use for upload pinata/ipfs
Following the ep-ui
And the return values are not the same structure

**Review by kesar — APPROVED** [review]

**Comment by kesar** [comment]
it needs to solve yarn.lock but everything else looks great 👍🏻 feel free to merge it once yarn.lock conflict is fixed!

**Pull request merged**